provider.go

```diff
@@ -37,6 +38,15 @@ func provider() terraform.ResourceProvider {
 					return "credential-store", nil
 				},
 			},
+			"profile": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: func() (interface{}, error) {
```

Comment:
If this should come from `AWS_PROFILE` environment variable that please use the `schema.MultiEnvDefaultFunc([]string{"AWS_PROFILE"}, "default")` default value provider so that environment variable is picked up by default.